modules/gapi/src/backends/cpu/gparsers.cpp

```diff
@@ -0,0 +1,332 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include "gparsers.hpp"
+
+namespace cv
+{
+namespace gapi
+{
+namespace parsers
+{
+class YoloParser
+{
+public:
+    YoloParser(const float* out, const int& side, const int& lcoords, const int& lclasses)
+        : m_out(out), m_side(side), m_lcoords(lcoords), m_lclasses(lclasses)
+    {}
+
+    float scale(const int& i, const int& b)
+    {
+        int obj_index = index(i, b, m_lcoords);
+        return m_out[obj_index];
+    }
+
+    double x(const int& i, const int& b)
+    {
+        int box_index = index(i, b, 0);
+        int col = i % m_side;
+        return (col + m_out[box_index]) / m_side;
+    }
+
+    double y(const int& i, const int& b)
+    {
+        int box_index = index(i, b, 0);
+        int row = i / m_side;
+        return (row + m_out[box_index + m_side * m_side]) / m_side;
+    }
+
+    double width(const int& i, const int& b, const float& anchor)
+    {
+        int box_index = index(i, b, 0);
+        return std::exp(m_out[box_index + 2 * m_side * m_side]) * anchor / m_side;
+    }
+
+    double height(const int& i, const int& b, const float& anchor)
+    {
+        int box_index = index(i, b, 0);
+        return std::exp(m_out[box_index + 3 * m_side * m_side]) * anchor / m_side;
+    }
+
+    float classConf(const int& i, const int& b, const int& label)
+    {
+         int class_index = index(i, b, m_lcoords + 1 + label);
+         return m_out[class_index];
+    }
+
+    cv::Rect toBox(const double& x, const double& y, const double& h, const double& w, const cv::Size& in_sz)
+    {
+        auto h_scale = in_sz.height;
+        auto w_scale = in_sz.width;
+        cv::Rect r;
+        r.x = static_cast<int>((x - w / 2) * w_scale);
+        r.y = static_cast<int>((y - h / 2) * h_scale);
+        r.width = static_cast<int>(w * w_scale);
+        r.height = static_cast<int>(h * h_scale);
+        return r;
+    }
+
+private:
+    const float* m_out = nullptr;
```

Comment:
So, I doubt a little now... should it be const or not? I insist only because in `SSDParser` the same thing is not the same.. So in my opinion these pointers (which are of the same purpose) should be treated in the same way (here and in `SSDParser`) Which way - you choose
Please, check this out

Reply:
The pointer is initialized with const float*. It must be the const float *.